File: 219. Contains Duplicate II/219. Contains Duplicate II.cs
public class Solution {
    public bool ContainsNearbyDuplicate(int[] nums, int k) {
        Dictionary<int,int> prev = new Dictionary<int,int>();
        int n = nums.Length;
        for(int i=0; i<n; i++){
            if(prev.ContainsKey(nums[i]) && Math.Abs(prev[nums[i]] - i)<=k){
                return true;
            }
            prev[nums[i]] = i;
        }

        return false;
    }
}

File: 219. Contains Duplicate II/219. Contains Duplicate II.cpp
class Solution {
public:
    unordered_map<int, int> prev;
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        int n = nums.size();
        for(int i=0; i<n; i++){
            if(prev.count(nums[i]) && abs(prev[nums[i]]-i)<=k) return true;
            prev[nums[i]] = i;
        }
        return false;
    }
};
